tests/python/test_atomic.py

```diff
@@ -35,18 +36,25 @@ def func():
         assert valproc(ya) == e
 
 
-ti.init(ti.cc, log_level=ti.DEBUG)
-run_atomic_add_global_case(ti.i32, 42)
+@ti.func
+def cas_atomic_add(dest: ti.template(), val):
+    old = dest
+    while ti.atomic_cas(dest, old, old + val) != old:
+        old = dest
+    return old
 
 
-@ti.all_archs
-def test_atomic_add_global_i32():
-    run_atomic_add_global_case(ti.i32, 42)
+@pytest.mark.parametrize('atomic_add', [ti.atomic_add, cas_atomic_add])
+@ti.test([ti.cc])
+def test_atomic_add_global_i32(atomic_add):
+    run_atomic_add_global_case(atomic_add, ti.i32, 42)
```

Comment:
nit: Will this remove the original `atomic_add` test on `all_archs` except `ti.cc`?